gulp-ng-config.js

```diff
@@ -30,10 +32,24 @@ function gulpNgConfig (moduleName, configuration) {
         jsonObj,
         wrapTemplate;
 
-    try {
-      jsonObj = JSON.parse(file.contents.toString('utf8'));
-    } catch (e) {
-      this.emit('error', new PluginError(PLUGIN_NAME, 'invaild JSON file provided'));
+    if (_.endsWith(file.path, 'yml') || _.endsWith(file.path, 'yaml')) {
+      configuration.parser = 'yml';
```

Comment:
Lastly -- I think we should do this *before* we apply the `defaults`. If we already have a value for our parser -- we want to always use that, and not override it by the file extension (the file extension is the fallback if the parser is not defined by the user's configuration -- before we apply a default configuration of `json`).

So ultimately we want to put this inbetween L26 and L27

```js
configuration = configuration || {};
if (!configuration.parser && (_.endsWith(file.path, 'yml') || ..) {
  configuration.parser = 'yml';
}
configuration = _.merge(...)
```